File: 3517-smallest-palindromic-rearrangement-i/3517-smallest-palindromic-rearrangement-i.java
class Solution {
    public String smallestPalindrome(String s) {
        char c[]=s.toCharArray();
        int n=c.length;
        Arrays.sort(c,0,n/2);
        // System.out.println(c);
        for(int i=0;i<n/2;i++){
            c[n-i-1]=c[i];
        }
        return new String(c);
            }
}

File: 3517-smallest-palindromic-rearrangement-i/3517-smallest-palindromic-rearrangement-i.py
class Solution:
    def smallestPalindrome(self, s: str) -> str:
        c=list(s)
        n=len(c)
        c[:n//2]=sorted(c[:n//2])
        for i in range (n//2):
            c[n-i-1]=c[i]
        return "".join(c)